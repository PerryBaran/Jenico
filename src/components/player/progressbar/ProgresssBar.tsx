import React, { RefObject, useEffect } from 'react';
import style from './progressBar.module.css';

interface Props{
    barRef: RefObject<HTMLInputElement>, 
    audioRef: RefObject<HTMLAudioElement>, 
    time:number, 
    duration: number
}

function ProgresssBar(props: Props) {
    const {barRef, audioRef, time, duration} = props;

    const changeTime = (value: number) => {
        if (audioRef.current) {
            audioRef.current.currentTime = value;
        }
    };
    
    useEffect(() => {
        if (barRef.current) {
            barRef.current.value = `${time}`
        }
    }, [time, barRef]);

    useEffect(() => {
        if (barRef.current) {
            barRef.current.max = `${duration}`;
        }  
    }, [duration, barRef]);

    return (
        <input 
            type='range'
            ref={barRef}
            name='time'
            min={0}
            defaultValue={0}
            onChange={e => changeTime(Number(e.target.value))}
            className={style.bar}/>
    );
}

export default ProgresssBar;