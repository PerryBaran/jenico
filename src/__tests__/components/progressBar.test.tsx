import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import ProgresssBar from "../../components/player/progressbar/ProgresssBar";

describe("ProgressBar", () => {
  const validProps: any = {
    time: 20,
    duration: 300,
    handleAudioTime: jest.fn(),
  };

  afterEach(cleanup);

  test("snapshot", () => {
    const { asFragment } = render(
      <ProgresssBar
        handleAudioTime={validProps.handleAudioTime}
        time={validProps.time}
        duration={validProps.duration}
      />
    );

    expect(asFragment).toMatchSnapshot();
  });

  test("content", () => {
    render(
      <ProgresssBar
        handleAudioTime={validProps.handleAudioTime}
        time={validProps.time}
        duration={validProps.duration}
      />
    );
    const input = screen.getByTestId("progress-bar");
    
    expect(input).toHaveAttribute("type", "range");
    expect(input).toHaveAttribute("name", "time");
    expect(input).toHaveAttribute("min", "0");
    expect(input).toHaveAttribute("max", validProps.duration.toString());
    expect(input).toHaveAttribute("value", validProps.time.toString());
  });

  test("input change", () => {
    render(
      <ProgresssBar
        handleAudioTime={validProps.handleAudioTime}
        time={validProps.time}
        duration={validProps.duration}
      />
    );
    const input = screen.getByTestId("progress-bar");
    const newValue = "40";

    expect(validProps.handleAudioTime).toBeCalledTimes(0);
    fireEvent.change(input, { target: { value: newValue } });
    expect(validProps.handleAudioTime).toBeCalledWith(Number(newValue));
  });
});