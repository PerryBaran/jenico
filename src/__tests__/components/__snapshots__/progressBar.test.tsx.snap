// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar snapshot 1`] = `[Function]`;
